src/poc/miner_poc_mgr.erl

```diff
@@ -0,0 +1,1001 @@
+                    Path = blockchain_poc_path_v4:build(TargetPubkeybin, TargetRandState, Ledger, Time, Vars),
+                    N = erlang:length(Path),
+                    [<<IV:16/integer-unsigned-little, _/binary>> | LayerData] = blockchain_txn_poc_receipts_v2:create_secret_hash(
+                        Entropy,
+                        N + 1
+                    ),
+                    OnionList = lists:zip([libp2p_crypto:bin_to_pubkey(P) || P <- Path], LayerData),
+                    {Onion, Layers} = blockchain_poc_packet_v2:build(Keys, IV, OnionList),
+                    [_|LayerHashes] = [crypto:hash(sha256, L) || L <- Layers],
+                    Challengees = lists:zip(Path, LayerData),
+                    PacketHashes = lists:zip(Path, LayerHashes),
+                    Secret = libp2p_crypto:keys_to_bin(Keys),
+                    %% save the POC data to our local cache
+                    LocalPOC = #local_poc{
+                        onion_key_hash = OnionKeyHash,
+                        block_hash = BlockHash,
+                        target = TargetPubkeybin,
+                        onion = Onion,
+                        secret = Secret,
+                        challengees = Challengees,
+                        packet_hashes = PacketHashes,
+                        keys = Keys,
+                        start_height = POCStartHeight
+                    },
+                    ok = write_local_poc(LocalPOC, State),
+                    lager:debug("started poc for challengeraddr ~p, onionhash ~p", [Challenger, OnionKeyHash]),
+                    ok
+            end
+    end.
+
+-spec process_block_pocs(
+    POCChallengeType :: libp2p_crypto:pubkey_bin(),
+    BlockHeight :: pos_integer(),
+    BlockHash :: blockchain_block:hash(),
+    BlockPOCs :: [{binary(), blockchain_ledger_poc_v3:poc()}],
+    Chain :: blockchain:blockchain(),
+    State :: state()
+) -> ok.
+process_block_pocs(
+    POCChallengeType,
+    BlockHeight,
+    BlockHash,
+    BlockPOCs,
+    Chain,
+    #state{chain = Chain} = State
+)  when POCChallengeType == validator ->
+    Ledger = blockchain:ledger(Chain),
+    lager:debug("poc mgr block pocs: ~p", [BlockPOCs]),
+    [
+        begin
+            OnionKeyHash = blockchain_ledger_poc_v3:onion_key_hash(POCV3),
+            %% use onion key hash to check our local cache containing the keys of POCs owned by this validator
+            %% if it is one of this local validators POCs, then kick it off
+            case cached_local_poc_key(OnionKeyHash) of
+                {ok, {_, #poc_local_key_data{keys = Keys}}} ->
+                    %% its a locally owned POC key, so kick off a new POC
+                    Vars = blockchain_utils:vars_binary_keys_to_atoms(
+                        maps:from_list(blockchain_ledger_v1:snapshot_vars(Ledger))),
+                    spawn(fun() -> initialize_poc(BlockHash, BlockHeight, Keys, Vars, State) end);
```

Comment:
This spawn doesn't track if anything crashed out, is that ok?

Reply:
yeah, thats fine at this point. If a POC fails to initiate there is no retry.  If we want to change that then we would need to reconsider